### PR TITLE
Parser Cleanup

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
This is an initial investigation of how some refactoring of the parser code, effects the perf.

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/Parser/ParserExtensions.vb [hunk 1]
```diff
@@ -16,6 +16,81 @@ Imports InternalSyntaxFactory = Microsoft.CodeAnalysis.VisualBasic.Syntax.Intern
 Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
     Friend Module ParserExtensions
 
+        <Extension>
+        Function IsIn(kind As SyntaxKind,
+                      kind0 As SyntaxKind, kind1 As SyntaxKind, kind2 As SyntaxKind, kind3 As SyntaxKind, kind4 As SyntaxKind,
+                      kind5 As SyntaxKind, kind6 As SyntaxKind, kind7 As SyntaxKind, kind8 As SyntaxKind, kind9 As SyntaxKind) As Boolean
+            Return (kind = kind0) Or (kind = kind1) Or (kind = kind2) Or (kind = kind3) Or (kind = kind4) Or
```
Comment (Youssef1313) [inline src/Compilers/VisualBasic/Portable/Parser/ParserExtensions.vb:23]: I think `OrElse` should perform better here.

File: src/Compilers/VisualBasic/Portable/Parser/ParserExtensions.vb [hunk 2]
```diff
@@ -16,6 +16,81 @@ Imports InternalSyntaxFactory = Microsoft.CodeAnalysis.VisualBasic.Syntax.Intern
...
+        End Function
+        <Extension>
+        Function IsIn(kind As SyntaxKind,
+                      kind0 As SyntaxKind, kind1 As SyntaxKind, kind2 As SyntaxKind, kind3 As SyntaxKind, kind4 As SyntaxKind,
+                      kind5 As SyntaxKind, kind6 As SyntaxKind, kind7 As SyntaxKind) As Boolean
+            Return (kind = kind0) Or (kind = kind1) Or (kind = kind2) Or (kind = kind3) Or (kind = kind4) Or
+                   (kind = kind5) Or (kind = kind6) Or (kind = kind7)
+        End Function
+                <Extension>
+        Function IsIn(kind As SyntaxKind,
```
Comment (Youssef1313) [inline src/Compilers/VisualBasic/Portable/Parser/ParserExtensions.vb:41]: Indentation isn't correct here.

File: src/Compilers/VisualBasic/Portable/Parser/ParserExtensions.vb [hunk 3]
```diff
@@ -16,6 +16,81 @@ Imports InternalSyntaxFactory = Microsoft.CodeAnalysis.VisualBasic.Syntax.Intern
...
+                   (kind = kind5) Or (kind = kind6) Or (kind = kind7) Or (kind = kind8) Or (kind = kind9)
+        End Function
+        <Extension>
+        Function IsIn(kind As SyntaxKind,
+                      kind0 As SyntaxKind, kind1 As SyntaxKind, kind2 As SyntaxKind, kind3 As SyntaxKind, kind4 As SyntaxKind,
+                      kind5 As SyntaxKind, kind6 As SyntaxKind, kind7 As SyntaxKind, kind8 As SyntaxKind) As Boolean
+            Return (kind = kind0) Or (kind = kind1) Or (kind = kind2) Or (kind = kind3) Or (kind = kind4) Or
+                   (kind = kind5) Or (kind = kind6) Or (kind = kind7) Or (kind = kind8)
+        End Function
+        <Extension>
```
Comment (Youssef1313) [inline src/Compilers/VisualBasic/Portable/Parser/ParserExtensions.vb:33]: Blank line after `End Function` makes the code more readable.

File: src/Compilers/VisualBasic/Portable/Parser/ParserExtensions.vb (hunk 1 above)
Comment (AdamSpeight2008) [reply src/Compilers/VisualBasic/Portable/Parser/ParserExtensions.vb:23]: Not in this case, as it compile to just `or` without any branches. "OrElse" would add conditional branch after each comparison.

**Pull request closed without merging**